app/stac_api/models.py

```diff
@@ -771,3 +773,77 @@ def update_asset_from_upload(self):
         self.asset.checksum_multihash = self.checksum_multihash
         self.asset.update_interval = self.update_interval
         self.asset.save()
+
+
+class CountBase(models.Model):
+    '''CountBase tables are used to help calculate the summary on a collection.
+    This is only performant if the distinct number of values is small, e.g. we currently only have
+    5 possible values for geoadmin_language.
+    For each assets value we keep a count of how often that value exists, per collection. On
+    insert/update/delete of an asset we only need to decrease and/or increase the counter value.
+    Since the number of possible values is small, the aggregate array calculation to update the
+    collection also stays performant.
+    '''
+
+    class Meta:
+        abstract = True
+
+    id = models.BigAutoField(primary_key=True)
+    collection = models.ForeignKey(
+        Collection,
+        related_name='+',
+        on_delete=models.CASCADE,
+    )
+    count = models.IntegerField(null=True, blank=True)
```

Comment:
what would be the use-case to have a value with null or blank count?
wouldn't it be better to not accept such situations and only store values having a count of 1 or bigger?